### ISSUE TITLE
Compilation problems on Windows 7 with MSys64 

### ISSUE THREAD

**Issue by Coastal0** [gimli-org/gimli#51]
Relatively new to this, so please excuse any dumb questions.

I am trying to install GIMLi via MSys, ultimately to convert GMsh meshes into BERT format.

Following the instructions from http://www.pygimli.org/installation.html, I seem to be coming unstuck at the makefile. 

I have hopefully attached the console feed  as a text file.

Both Mercurial and Python 3.5.2 are installed (and apparently working, but I'm not totally sure).

> $ hg
> Mercurial Distributed SCM
> 
> basic commands:
> 
>  add           add the specified files on the next commit
>  annotate      show changeset information by line for each file
> ...
> (use "hg help" for the full list of commands or "hg -v" for details)

If there's any advice I'd really appreciate a hand to set this up...!

[console_out.txt](https://github.com/gimli-org/gimli/files/718590/console_out.txt)

**Comment by halbmy** [comment]
I will try to reproduce this with a fresh Windows 10 PC next week.
MSYS recently updated GCC to version 6.3 and llvm/clang to 3.9, and I guess this could cause the problems. I can provide you with a binary Windows installer for Python 3.5 if you like.

**Comment by carsten-forty2** [comment]
its a little strange .. I actually tried win7@vbox and there were some issues while bringing mingw up to date .. In the end I needed a fresh mingw installation with multiple update calls as described here (http://msys2.github.io/) .. but then everything went smooth with:

```curl -Ls install.pygimli.org | bash```

If the hg problem persists .. where your HG comes from ```which hg``` ?

**Comment by halbmy** [comment]
As there was no response from Alex after Carstens last mail for months and things might have changed anyway, I am closing this issue. @Coastal0 if there are still problems, just reopen the issue and provide an update.

**Comment by Coastal0** [comment]
Sorry guys! I had forgotten about this, and other things managed to take preference...

I have just started up again, and have managed to get exactly nowhere. :(
I am still missing the HG part.

```
$ which hg
/usr/bin/hg
```

```
$ pacman -Ss mercurial
msys/mercurial 4.0.2-1 (VCS) [installed]
    A scalable distributed SCM tool
```

`pacman -Sy mercurial`

```
==========================================
Installing gimli for win64
------------------------------------------
~/gimli ~
~/gimli/gimli ~/gimli ~
Already up-to-date.
~/gimli ~
CLEAN,
~/gimli/build ~/gimli
-- The C compiler identification is GNU 6.3.0
-- The CXX compiler identification is GNU 6.3.0
-- Check for working C compiler: C:/msys64/mingw64/bin/cc.exe
-- Check for working C compiler: C:/msys64/mingw64/bin/cc.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: C:/msys64/mingw64/bin/c++.exe
-- Check for working CXX compiler: C:/msys64/mingw64/bin/c++.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Target is 64 bits
-- ThirdParty set to: C:/msys64/home/264401k/gimli/gimli/../thirdParty/
-- External set to: C:/msys64/home/264401k/gimli/thirdParty/dist-GNU-6.3.0-64
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - found
-- Found Threads: TRUE
-- Found PythonInterp: C:/Anaconda3/python.exe (found version "3.5.2")
-- BOOST_ROOT does not exist:
-- Boost version: 1.64.0
-- Boost is local False
-- Boost found
-- BOOST_ROOT does not exist:
-- BOOST_ROOT,
-- Boost_INCLUDE_DIR, C:/msys64/mingw64/include
-- Boost version: 1.64.0
-- Found the following Boost libraries:
--   system
--   thread
--   chrono
--   date_time
--   atomic
-- boost include: C:/msys64/mingw64/include
-- Looking for C++ include boost/interprocess/managed_shared_memory.hpp
-- Looking for C++ include boost/interprocess/managed_shared_memory.hpp - found
-- Looking for C++ include boost/bind.hpp
-- Looking for C++ include boost/bind.hpp - found
-- HAVE_BOOST_INTERPROCESS_MANAGED_SHARED_MEMORY_HPP 1 C:/msys64/mingw64/include
-- Checking for package 'TRIANGLE'
-- External dir: C:/msys64/home/264401k/gimli/thirdParty/dist-GNU-6.3.0-64
-- Triangle_INCLUDE_DIR = Triangle_INCLUDE_DIR-NOTFOUND
-- Triangle_LIBRARIES = Triangle_LIBRARIES-NOTFOUND
-- Triangle could not be found. (missing:  Triangle_LIBRARIES Triangle_INCLUDE_DIR)
-- Triangle is local True
-- Found Wget: C:/msys64/usr/bin/wget.exe
-- Found Subversion: C:/msys64/usr/bin/svn.exe (found version "1.9.5")
-- Found Git: C:/msys64/usr/bin/git.exe (found version "2.13.0")
CMake Error at C:/msys64/mingw64/share/cmake-3.8/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find Hg (missing: HG_EXECUTABLE)
Call Stack (most recent call first):
  C:/msys64/mingw64/share/cmake-3.8/Modules/FindPackageHandleStandardArgs.cmake:377 (_FPHSA_FAILURE_MESSAGE)
  C:/msys64/mingw64/share/cmake-3.8/Modules/FindHg.cmake:94 (find_package_handle_standard_args)
  cmake/GIMLImacros.cmake:152 (find_package)
  cmake/GIMLImacros.cmake:193 (findBuildTools)
  cmake/GIMLImacros.cmake:169 (find_or_build_package_check)
  CMakeLists.txt:215 (find_or_build_package)


-- Configuring incomplete, errors occurred!
See also "C:/msys64/home/264401k/gimli/build/CMakeFiles/CMakeOutput.log".
make: *** No targets specified and no makefile found.  Stop.
~/gimli

```

**Comment by Coastal0** [comment]
I don't think I can re-open this issue either, sorry...!

**Comment by florian-wagner** [comment]
Hi @Coastal0,
it looks like you are trying to compile pygimli with Anaconda? At least the Python interpreter `C:\Anaconda3\python` is chosen. I think nobody has tried this so far (and a simple `conda install pygimli` is not yet possible on Windows), but generally it should work. If you decide to do so, however, I recommend to install all dependencies via `conda`. So for `hg`, you would do a `conda install mercurial`, etc. This is most (and likely only) important for boost, since boost_python needs to match the python version you are using. In your case you using Anaconda python but your boost comes from `C:/msys64/mingw64/include`. So again, a `conda install boost` seems to be necessary. Alternatively you can try our new Windows Installer (Based on WinPython 3.5, not Anaconda!), which you can download here: http://www.resistivity.net/download/setup-pygimli1.0.0RC1win64py35.exe

Let me know how it works out.

Cheers
Florian

**Comment by Coastal0** [comment]
Conda threw up all sorts of errors when I tried to update it just now, but most notably, a conflict with mercurial for python 2.7.

I'll give it a shot with your winpython, cheers!

**Comment by halbmy** [comment]
I hope the shot was successful. 
Windows installers will be available from the upcoming version 1.0 on for Python 3.5 and 3.6. Note that they work with both WinPython and Anaconda distributions.
